sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/ZE_experimental_bindless_images.asciidoc

```diff
@@ -0,0 +1,621 @@
+///       + ::zeMemAllocPitchedExp
+///         - ::rowPitch and ::slicePitch are [out] parameters
+///       + ::ze_image_desc_t.pNext
+///         - ::rowPitch and ::slicePitch are [in] parameters
+///         - When this structure is passed to an image descriptor,
+///           ::zeImageCreate must not allocate any additional device memory for
+///           the created image, only return a handle.
+///         - Additionally, when ::zeImageDestroy is called on an image created
+///           from a ::ze_image_mem_alloc_exp_desc_t allocation,
+///           ::zeImageDestroy must not free the memory associated with the
+///           image handle.
+///
+typedef struct _ze_image_usm_alloc_exp_desc_t {
+    ze_structure_type_t stype;      // [in]
+    void *pNext;                    // [in,out][optional]
+    size_t widthInBytes;            // [in]
+    size_t height;                  // [in][optional]
+    size_t depth;                   // [in][optional]
+    uint32_t elementSizeBytes;      // [in]
+    size_t rowPitch;                // [in,out]
+    size_t slicePitch;              // [in,out]
+} ze_image_usm_alloc_exp_desc_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Describes device specific restrictions for image properties
+///
+/// @details
+///     - This structure is passed to:
+///       + ::ze_device_properties_t::pNext
+///
+typedef struct _ze_device_image_bindless_exp_desc_t {
+    ze_structure_type_t stype;      // [in]
+    void *pNext;                    // [in,out][optional]
+    uint32_t imagePitchAlign;       // [out]
+    size_t maxImageLinearWidth;     // [out]
+    size_t maxImageLinearHeight;    // [out]
+    size_t maxImageLinearPitch;     // [out]
+} ze_device_image_bindless_exp_desc_t;
+```
+
+=== Interfaces
+
+```cpp
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Allocates on device memory intended for images
+///
+/// @details
+///     - Allocates on device image memory in an implementation specific layout and encoding
+///     - The memory allocated through this function should not be freed with ::zeMemFree,
+///     - but rather with ::zeMemFreeImageExp
+///     - Copy operations to or from this memory need to use ::zeCommandListAppendImage
+///
+/// @returns
+///     - ::ZE_RESULT_SUCCESS
+///     - ::ZE_RESULT_ERROR_INVALID_ARGUMENT
+///         + invalid image_desc
+///         + invalid mem_handle
+///     - ::ZE_RESULT_ERROR_OUT_OF_DEVICE_MEMORY
+///         + out of memory
+///     - ::ZE_RESULT_ERROR_INVALID_NULL_HANDLE
+///         + nullptr == hContext
+ZE_APIEXPORT ze_result_t ZE_APICALL
+zeMemAllocImageExp(
+    ze_context_handle_t hContext,                       /// [in] handle of the context object
+    const ze_image_mem_alloc_exp_desc_t *image_desc,    /// [in] memory allocation descriptor
+    ze_image_mem_handle_exp_t *mem_handle               /// [out] pointer to device allocation handle
+    );
```

Comment:
I dont think we need this function. We could just reuse zeImageCreate, and through pNext, pass ze_image_mem_alloc_exp_desc_t. When L0 detects that descriptor in pNext, then it would behave as this new one. That way we minimize the number of zeImageCreate APIs in L0  - we already have 2, this would be the 3rd - and that way we minimize confusion. @wdamon-intel, what do you think?

Reply:
thanks @isaacault for the details. Yes, it seems weird to call twice zeImageCreate, so makes more sense to have it separate.